Unique name for auth file

finkandreas opened the pull request:
This should fix #30

Madeeks commented:
Thanks for opening this!
A few kind requests and suggestions:
- Could you please rebase this branch on `develop` and also target the PR to `develop`?
- Would you be willing to update the [unit tests regarding the auth file](https://github.com/eth-cscs/sarus/blob/develop/src/image_manager/test/test_SkopeoDriver.cpp#L216)?
- The utility function [makeUniquePathWithRandomSuffix()](https://github.com/eth-cscs/sarus/blob/develop/src/common/Utility.cpp#L434) is a more fitting choice for this fix, and internally calls `generateRandomString()` which you already used in the commit.
- I believe this deserves a mention as a `Fixed` item in the changelog. I can take care of that if you prefer.

finkandreas committed:
Unique name for auth file

finkandreas commented:
- I have changed the base to `develop`
- adapted the unit tests
  - The first check for non-existent `auth.json` seems a little bit unnecessary, but I left it there
- I did not spot the functio n`makeUniquePathWithRandomSuffix()`, but you are absolutely right, it is the better choice
- I added a changelog item, feel free to change the wording there

Madeeks commented on src/image_manager/SkopeoDriver.cpp:
```diff
@@ -230,11 +230,12 @@ void SkopeoDriver::acquireAuthFile(const common::Config::Authentication& auth, c
 
     common::createFoldersIfNecessary(authFileBasePath);
-    authFilePath = authFileBasePath / "auth.json";
+    authFilePath = common::makeUniquePathWithRandomSuffix(authFileBasePath / "auth.json");
```
Given the way `makeUniquePathWithRandomSuffix()` works, the auth file path would result in `<base path>/auth.json-<random string>`.

Do you think something like
`common::makeUniquePathWithRandomSuffix(authFileBasePath / "sarus-auth").replace_extension(.json);`
would be more elegant?

This is a pure formality, since the code would work anyways (considering the path is passed explicitly to Skopeo).

Madeeks commented on src/image_manager/test/test_SkopeoDriver.cpp:
```diff
@@ -242,7 +242,9 @@ TEST(SkopeoDriverTestGroup, acquireAuthFile) {
             CHECK_FALSE(boost::filesystem::exists(expectedAuthFilePath));
```
With the auth file moving to a random path, it makes little sense to check whether a hardcoded path exists or not. I agree this is unnecessary and can be removed.

Madeeks commented on src/image_manager/test/test_SkopeoDriver.cpp:
```diff
@@ -242,7 +242,9 @@ TEST(SkopeoDriverTestGroup, acquireAuthFile) {
             auto driver = image_manager::SkopeoDriver{config};
-            driver.acquireAuthFile(config->authentication, config->imageReference);
+            auto randomExpectedAuthFilePath = driver.acquireAuthFile(config->authentication, config->imageReference);
+            CHECK_TRUE(expectedAuthFilePath != randomExpectedAuthFilePath);
```
Writing  here but the following comments apply to all  cases under this test.

I think the check and reassignment between `expectedAuthFilePath`  and `randomExpectedAuthFilePath` is not necessary and can be simplified to a direct assignment to `authFilePath`.

On the other hand, I suggest to check that the auth file is created in the expected directories (in the case of this code scope, the XDG Runtime Dir). For example, something like:
```
CHECK(authFilePath.parent_path() == (xdgRuntimeDir / "sarus"))
```

finkandreas replied:
Done - I think I kept the check that the filename is unique, to make sure that I really do not use the old implementation. Anyway I removed it now, since I'm almost certain that it does not create the old static filename.

finkandreas committed:
have a json suffix for unique auth file
test if auth file is in the expected directory

Madeeks reviewed (approved):
LGTM!

Madeeks committed:
Updated contributors list
